mantidproject/mantid
# Add plot setting to hide/show legend by default

**Description of work.**
This PR adds a setting to the workbench plot settings to allow users to hide/show the legend by default.

**To test:**
- Run workbench and load some data
- Go to settings and plots
- There's a tickbox (Show legend) which is ticked by default
- Turn it off and plot a spectra, there should be no legend
- Likewise if you turn it back on and plot a spectra there should be a legend



Fixes #28253 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
